docs: Add Tweets API endpoint documentation (verified against 7.0) (#479)
mautic/developer-documentation-new: docs/rest_api/tweets.rst

> Tweets

Reviewer: Content migrated from `source/includes/_api_endpoint_tweets.md` in the legacy developer-documentation repo. Converted Markdown to RST, updated formatting to match existing `rest_api/*.rst` conventions (list-table directives, vale on/off, cross-references), and modernized example timestamps.

Source: https://github.com/mautic/developer-documentation/blob/main/source/includes/_api_endpoint_tweets.md

> Tweets

Reviewer: Verified all Tweet entity properties against `Tweet.php` in Mautic 7.0 branch. Added missing fields (`description`, `mediaId`, `mediaPath`, `sentCount`, `favoriteCount`, `retweetCount`), confirmed 280-character text limit, and removed non-existent `tweetId` field from JSON examples.

Source: https://github.com/mautic/mautic/blob/7.0/plugins/MauticSocialBundle/Entity/Tweet.php

######

Use this endpoint to obtain details on Mautic's Tweets. Implemented in Mautic 2.8.0.

.. note::

>    The Tweets feature is part of the MauticSocialBundle plugin. Twitter integration must be enabled for this API to work.

Reviewer 2: Let's address Vale warnings here.

```suggestion
   The Tweets feature is part of the **Mautic Bundle for Social** Plugin. Integration with Twitter must be active for this API to work.
```

   * - ``dateModified``
     - datetime
     - Tweet record last modification date and time
   * - ``modifiedBy``
     - integer
     - ID of the User who last modified the Tweet
   * - ``modifiedByUser``
     - string
     - Name of the User who last modified the Tweet
   * - ``language``
     - string
     - The language code for the Tweet, such as ``en``, ``fr``, and so on
   * - ``category``
     - object
     - The Category for the Tweet
   * - ``mediaId``
     - string
     - ID of the Twitter media object attached to the Tweet
   * - ``mediaPath``
     - string
     - Path to the local media file
   * - ``sentCount``
     - integer
     - Number of times this Tweet has been sent
   * - ``favoriteCount``
     - integer
     - Number of favorites or likes on the Tweet
   * - ``retweetCount``
     - integer
     - Number of retweets

.. vale on

.. vale off

List Tweets
***********

.. vale on

Retrieves a list of Tweets.

.. code-block:: php

   <?php
   // ...

   $tweets = $tweetApi->getList($searchFilter, $start, $limit, $orderBy, $orderByDir, $publishedOnly, $minimal);

.. vale off

HTTP request
============

.. vale on

``GET /tweets``

Query parameters
----------------

> 

Reviewer 2: As it "published" is intentional here, let's wrap the table in vale off/on statements.

```suggestion

.. vale off

```

.. list-table::
   :widths: 20 20 60
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``search``
     - string
     - String or search command to filter entities
   * - ``start``
     - integer
     - Starting row for the returned entities - defaults to 0
   * - ``limit``
     - integer
     - Maximum number of entities to return - defaults to 30
   * - ``orderBy``
     - string
     - Column to sort by. Any column in the response is valid.

       **Note**: convert ``camelCase`` properties to ``snake_case``. For example, ``dateAdded`` becomes ``date_added``, and so on
   * - ``orderByDir``
     - string
     - Order direction - ``asc`` or ``desc``
   * - ``publishedOnly``
     - boolean
     - Returns only currently published entities
   * - ``minimal``
     - boolean
     - Returns only a simple mapped object of entities without additional lists in it

> 

Reviewer 2: ```suggestion

.. vale on

```

               "createdBy": 1,
               "createdByUser": "John Doe",
               "modifiedBy": 1,
               "modifiedByUser": "John Doe",
               "id": 1,
               "name": "Thank you tweet",
               "text": "Hi {twitter_handle}\n\nThanks for ...",
               "description": "Used in the Product A campaign 1",
               "language": "en",
               "category": null,
               "mediaId": null,
               "mediaPath": null,
               "sentCount": 3,
               "favoriteCount": 0,
               "retweetCount": 0
           }
       ]
   }

Properties
----------

.. list-table::
   :widths: 25 25 50
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``total``
     - integer
     - Total count of Tweets
   * - ``tweets``
     - array
     - An array of Tweet objects

.. vale off

For the rest of the properties, refer to :ref:`Tweet properties <get Tweet properties>`.

.. vale off

Create Tweet
************

.. vale on

Creates a new Tweet.

.. code-block:: php

   <?php

   $data = array(
       'name' => 'Tweet A',
       'text' => 'This is my first tweet created via API.',

>    );

Reviewer 3: Let's use modern syntax
```suggestion
   $data = [
       'name' => 'Tweet A',
       'text' => 'This is my first tweet created via API.',
   ];
```

     - The language code for the Tweet, such as ``en``, ``fr``, and so on
   * - ``category``
     - integer
     - ID of the Category for the Tweet
   * - ``asset``
     - integer
     - ID of an Asset to link to the Tweet
   * - ``page``
     - integer
     - ID of a Page to link to the Tweet

.. vale on

Response
========

* Returns ``201 Created`` when the request successfully creates a Tweet.

The response is a JSON object similar to :ref:`Get Tweet <get Tweet response>`.

Properties
----------

Refer to :ref:`Tweet properties <get Tweet properties>`.

.. vale off

Edit Tweet
**********

.. vale on

Edits a Tweet.

This operation supports ``PUT`` or ``PATCH`` depending on the desired behavior:

* ``PUT``: **full replacement**. The request creates a new Tweet if the ID is missing. If the ID exists, the request clears all existing data and replaces it with the provided values.
* ``PATCH``: **partial update**. The request only updates field values based on the request data. The request fails when the Tweet ID doesn't exist.

.. code-block:: php

   <?php

   $id   = 1;
   $data = array(
       'name' => 'Tweet A',
       'text' => 'This is my first tweet created via API.',

>    );

Reviewer 3: ```suggestion
   $data = [
       'name' => 'Tweet A',
       'text' => 'This is my first tweet created via API.',
   ];
```

Tweets
######

> Use this endpoint to obtain details on Mautic's Tweets. Implemented in Mautic 2.8.0.

Reviewer 2: @escopecz I have questions:

- Is it true that this API only implemented in version 2.8.0? 
- Is the wording for the 1st sentence correct? Would "Use this endpoint to obtain Tweets details.: more accurate?

TIA! ✨

Reviewer 3: I think we can drop the information about the version when this was introduced. Not relevant anymore in version 7. 

Or we can say Tweet Entities or something like it. But all is understandable. Whatever you like.
